### ISSUE TITLE
macOS Sequoia and tvOS 18 support?

### ISSUE THREAD

**Issue by nimrodbens** [bsharper/atv-desktop-remote#31]
the app stopped working for me since i updated both operating systems.

it's stuck here: 

<img width="308" alt="Screenshot 2024-09-20 at 13 24 56" src="https://github.com/user-attachments/assets/347e0f76-9ed4-4ff2-8454-7eaca2c6cb31">

i tried deleting the `ATV Remote` folder and restart the app. still doesn't work.

**Comment by stayingarmed** [comment]
Same issue and sadly I have a feeling it probably won't be fixed because it seems the program is abandoned

**Comment by bsharper** [comment]
It's not abandoned, it's just a shared personal project. I made this for me.

Things you can try: 

- Install command line tools: `xcode-select --install`

- Install homebrew: https://brew.sh

- Install pyatv directly https://pyatv.dev/documentation/#latest-stable-version

Pyatv is how this program communicates with the Apple TV. That plus websockets are the only Python packages it uses, that's what it's doing when that screen shows up.

- You can also look at the setup code and try running it line by line to find out where it stops for you: https://github.com/bsharper/atv-desktop-remote/blob/main/build/start_server.sh

There are other closed issues you might have to look through, but I'm sure you can get it working. Generalizing every issue into a robust setup file is hard, but I'm open to pull requests.

**Comment by wesgurn** [comment]
I had the same issue when I updated to macOS Sequoia.  To fix the issue, go into the ~/Library/Application Support/ATV Remote/env/bin and run **./atvremote wizard** 

This seemed to reset the pairing and I could repair it again using atv remote.

**Comment by roar5138** [comment]
I tried most of these fixes unfortunately they didn't work. I hope you are able to update this program soon. It was such a nice tool to use around the house. I was searching for something like this for years and just found your program a couple months ago. Too short a time to use it.

**Comment by bsharper** [comment]
@roar5138 Can you share the file `atv_pip_install.log` from the path ~/Library/Application Support/ATV Remote? You can find this by going to Finder then pressing Cmd+Shift+G then pasting `~/Library/Application Support/ATV Remote` and pressing enter. There should be a file called `atv_pip_install.log`. If you post that file I might be able to see what is causing the initialization to fail.

**Comment by roar5138** [comment]
Unfortunately, I don't see that specific log file. I have uninstalled and re-installed it a couple times before remembering there was a TVos Update.  This is what the folder looks like if you have any ideas.
Thanks for getting back to me so soon. 


[Screen Recording 2024-10-10 at 11.24.58 AM.mov.zip](https://github.com/user-attachments/files/17331356/Screen.Recording.2024-10-10.at.11.24.58.AM.mov.zip)

**Comment by bsharper** [comment]
If you are running Sequoia, can you make sure this setting is correct under Privacy and Security / Local Network?
<img width="468" alt="image" src="https://github.com/user-attachments/assets/667426d0-28a6-4244-8b60-65f55e7f5847">

If it's not, I'd try enabling it and then deleting ~/Library/Application Support/ATV Remote (just drag ATV Remote to the trash can) and starting it again. The program will recreate this directory and everything in it, including the `atv_pip_install.log` file.

**Comment by nimrodbens** [comment]
just to chime in, ever since i created this issue the app went back to working normally again. i didn't change anything in regard to the two operating systems. if my logs can help in any way i would be happy to post them.

**Comment by roar5138** [comment]
@bsharper Yeah I had that setting on my local network. Thanks for all the help everyone. Maybe one day my will return to normal like @nimrodbens Hah, glad you got it back! I'll keep trying and if you have an idea, let me know!

**Comment by roar5138** [comment]
hey all, just trying again to maybe get this back. am I missing anything as im looking for something called "./atvremotewizard". Just imagine you're talking to someone who definitely isn't a programmer....

<img width="647" alt="Screenshot 2024-10-18 at 12 26 37 PM" src="https://github.com/user-attachments/assets/303203ca-84a5-4196-88af-15c4fcb9d887">

**Comment by bsharper** [comment]
@roar5138 I'm guessing you don't have command line tools installed? Seems to be starting and stopping quickly. If you had them installed before you might have to install them again: https://mac.install.guide/commandlinetools/4

Or the short version, type `xcode-select –-install` in the terminal.

**Comment by Doty1154** [comment]
> If you are running Sequoia, can you make sure this setting is correct under Privacy and Security / Local Network? <img alt="image" width="468" src="https://private-user-images.githubusercontent.com/3291225/375467984-667426d0-28a6-4244-8b60-65f55e7f5847.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQwODkwMzksIm5iZiI6MTczNDA4ODczOSwicGF0aCI6Ii8zMjkxMjI1LzM3NTQ2Nzk4NC02Njc0MjZkMC0yOGE2LTQyNDQtOGI2MC02NWY1NWU3ZjU4NDcucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIxMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMTNUMTExODU5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NDMyNjEwNTQ2MGFhMjc4OTM2ZDM4MTcyMmE2YzE0ZjEwZTVjZWI0NDhkZWM1M2ZmY2ExYTAxODMzM2Q0YTVhZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ._FoKpXLoRPEtK0lvXSuYsgv8mQn0mjpRmuAGHQLk8T4">
> 
> If it's not, I'd try enabling it and then deleting ~/Library/Application Support/ATV Remote (just drag ATV Remote to the trash can) and starting it again. The program will recreate this directory and everything in it, including the `atv_pip_install.log` file.

Deleting the folder fixed the app for me, thanks for the app : )

**Comment by paulmc49255** [comment]
I found the solution to this problem after diving into it myself on MacOS (AMD).  Python uses Xcode and you have to accept the terms and conditions of it.  I also updated Homebrew. 

1- Close out of the ATV application.  
2-Delete the folder `~/Library/Application Support/ATV Remote`
3-Open Terminal on MacOS
4-In Terminal type `cd /`
5-In Terminal type `sudo xcodebuild -license accept`
  -Terminal will ask you for your MacOS password
6-In Terminal type `brew update`
7-After Terminal is done with the tasks, restart your Mac
8-Launch the ATV application

**Comment by TheBaconbitz** [comment]
> Thanks everyone for the help. I just kept deleting the application support folder and the program and reinstalling. Been doing this on and off for the last couple months but it finally started to work again! Just had the program running in the background and eventually (mighta had to do with a new ATV update) it clicked. Very glad to have it back.

This helped me too. Thanks.
